policy/kubernetes/matches.rego

```diff
@@ -1,6 +1,10 @@
 package k8s
 import data.kubernetes
 
+matches[[kind, namespace, name, resource]] {
+	resource := kubernetes[kind][namespace][name].object
```

Comment:
Why is this needed ?

Reply:
I changed the call from the /admit endpoint to the OpenPolicyAgent to add an additional layer. The original resource is send as Object and things UserInfo or OldObject are send with it. So a call to the OPA looks like the following now:
````
data.admission.deny[
  {
    "id": id,
    "resource": {
      "kind": "namespaces",
      "name": "CCbwwpbqch"
    },
    "resolution": resolution
  }
] with data["kubernetes"]["namespaces"]["CCbwwpbqch"] as {
  "uid": "1f5ea330-ec88-11e8-8184-fa163ec0ee41",
  "kind": {
    "group": "",
    "version": "v1",
    "kind": "Namespace"
  },
  "resource": {
    "group": "",
    "version": "v1",
    "resource": "namespaces"
  },
  "operation": "CREATE",
  "userInfo": {
    "username": "cluster-admin",
    "groups": [
      "system:masters",
      "system:authenticated"
    ]
  },
  "object": {
    "metadata": {
      "name": "test",
      "uid": "1f5e9e79-ec88-11e8-8184-fa163ec0ee41",
      "creationTimestamp": "2018-11-20T05:50:02Z"
    },
    "spec": {
      "finalizers": [
        "kubernetes"
      ]
    },
    "status": {
      "phase": "Active"
    }
  }
},
````
This has the advantage that Validating/MutatingWebhooks can also decide based on UserInfo or the oldObject (if that makes sense). Unfortunately the normal machtes does not like before. So I've added another matches: one works now for the /audit scenario and one for /admit and the rules defined in .rego files work independent of its an /admit or /audit use case and do not have to be changed.

Reply:
Makes perfect sense thanks was planing to do that :). Why do you still have the old matches in the package ?

Reply:
Because I thought kube-mgmt still syncs the Kubernetes resources with the "old" format, so that the rules work for both. But actually I'm not sure ;)